src/handlers/handle-scheduled-checkin.ts

```diff
@@ -1,40 +1,51 @@
+import * as Sqs from '@aws-sdk/client-sqs';
+import assert from 'assert';
 import console from 'console';
 import * as Got from 'got';
 import * as Luxon from 'luxon';
+import process from 'process';
 import * as util from 'util';
 import * as CheckIn from '../lib/check-in';
-import * as EventDetail from '../lib/event-detail';
+import * as Queue from '../lib/scheduled-checkin-ready-queue';
 import * as SwClient from '../lib/sw-client';
 import * as SwGenerateHeaders from '../lib/sw-generate-headers';
 
 /**
- * On scheduled checkin, check the user in.
- *
- * @todo also remove the associated eventBridge rule and eventBridge trigger
+ * Check the user into a single flight.
  */
-export async function handle(event: EventDetail.Detail) {
-  let output: unknown;
-
+export async function handle(event: AWSLambda.SQSEvent) {
   try {
-    output = await handleInternal(event);
+    await handleInternal(event);
   } catch (error) {
     console.error(error);
+
+    // allow sending to dlq immediately
+    await adjustMessageVisibilityTimeout({
+      queueUrl: process.env.SCHEDULED_CHECKIN_READY_QUEUE_URL,
+      receiptHandle: event.Records[0].receiptHandle,
+      visibilityTimeout: 0
+    });
+
+    // send to dlq
     throw error;
```

Comment:
What ends up going to the DLQ? The error itself?

Reply:
The same message that this function received. So if we want to implement notifications, we should add the user's email address and/or password to the EventBridge rule's detail object, which will end up in the SQS message, which will be delivered to the DLQ consumer.